fontforge/fvfonts.c

```diff
@@ -1304,40 +1307,42 @@ bool InterpolationSanity(SplineSet* base, SplineSet *other, int order2, char* pr
         int bicw = SplinePointListIsClockwise(bss), oicw = SplinePointListIsClockwise(oss);
 
         if (bicw == -1 || oicw == -1)
-            TRACE(_("%sInterpolationSanity: Assuming self-intersecting contour %d to be sane"), prefix, contour_count);
+            TRACE(_("%sInterpolationSanity: Assuming contour %d of undefined direction to be sane\n"), prefix, contour_count);
         else if (bicw != oicw) {
             LogError(_("%sIncompatible contour directionality (contour %d)"), prefix, contour_count);
             ret = false;
         }
 
-        int bpnum = SplineSetNumberPoints(bss, order2);
-        int opnum = SplineSetNumberPoints(oss, order2);
-        if (bpnum != opnum) {
-            LogError(_("%sIncompatible point counts: %d vs %d (countour %d)"), prefix, bpnum, opnum, contour_count);
-            ret = false;
-        }
-
         int segment_count = 1;
         for ( SplinePoint *bsp=bss->first, *osp=oss->first; ; segment_count++ ) {
-            bool bnonextpt = bsp->next==NULL;
-            bool ononextpt = osp->next==NULL;
-            if (bnonextpt != ononextpt) {
-                char* missingnextpt = bnonextpt ? _("base") : _("other");
-                LogError(_("%sWe expected another point in the %s spline after segment %d"), prefix, missingnextpt, segment_count);
-                ret = false; break;
-            }
-            bool bcurve = ( !bsp->nonextcp || (!bnonextpt && !bsp->next->to->noprevcp) );
-            bool ocurve = ( !osp->nonextcp || (!ononextpt && !osp->next->to->noprevcp) );
+            bool bcurve = ( !bsp->nonextcp || (bsp->next!=NULL && !bsp->next->to->noprevcp) );
+            bool ocurve = ( !osp->nonextcp || (osp->next!=NULL && !osp->next->to->noprevcp) );
+
             if ( bcurve != ocurve ) {
                 char* wrongpttype = bcurve ? _("base") : _("other");
                 LogError(_("%sSegment %d in the %s spline is a curve when it should be a line"), prefix, segment_count, wrongpttype);
-                ret = false; break;
+                ret = false;
             }
-            if ( bnonextpt ) break;
-            bsp = bsp->next->to;
-            osp = osp->next->to;
+
```

Comment:
The code below seems strangely organized to me. I would do something like:

```
           // This assumes you're verifying both are closed or open up front, which is worth doing to give it its own error message. 
            bool bnonextpt = ( bsp->next==NULL || bsp->next->to==bss->first );
            bool ononextpt = ( osp->next==NULL || osp->next->to==bss->first );

            if ( bnonextpt && ononextpt ) 
                  break;
            if ( bnonextpt != ononextpt ) {
                   char *problemspline = bnonextpt ? _("base") : _("other");
                   /* extra point count and error here */
                   ret = false;
                   break;
             }
             bsp = bsp->next->to;
             osp = osp->next->to;
```
My major substantial argument for this is that lines 1329 and 1330 selectively advance the individual pointers before the remaining checks and that's just really confusing.